frog/imports/api/users.js

```diff
@@ -0,0 +1,50 @@
+// @flow
+import { Meteor } from 'meteor/meteor';
+
+type User = {
+  id?: string,
+  userObj?: mixed
+};
+
+/**
+ * Returns the appropriate username based on the type of user. If no user is passed then will return the current user's username
+ *
+ * @param: {User=} user
+ */
+
+export const getUsername = (user: User) => {
+  const selectedUser = getUser(user);
+  if (selectedUser) {
+    if (selectedUser.isAnonymous) return 'Anonymous User';
+    else if (selectedUser.emails) return selectedUser.profile.displayName;
+    else if (selectedUser.username) return selectedUser.username;
+  }
+  return 'No user logged in';
+};
+
+/**
+ * Returns the appropriate user object based on the type of user. If no user is passed as args then will return the current user object.
+ *
+ * @param: {User=} user
+ */
+const getUser = (user: User) => {
```

Comment:
I don't really understand this function:
- in the case the user has an id: we return a user that has the id. Couldn't we simply have a function called findUserById? In which scenario would we have an id but no userObj?
- in the case the user has an obj but no id: We already have the obj, why do we need this function?
- in the case we have nothing: we return the current user. Would suggest creating a function called getCurrentUser

Reply:
This is a helper function for the main function (note that it is not exported), which makes it easier to call the main function in three different ways (with current user, with id, with object). I asked for this, and I think it's OK.

Reply:
The idea behind this API was to  get the correct display name for each type of user by either passing in a user object or a userid and thus reduce the number of checks we have and also the number of calls to Meteor.user().username. 
getUser is not being exported but just as a helper to getUsername. In different parts of the FROG codebase we want to get the username by either passing the user id or user object or nothing.